modules/SofaBoundaryCondition/ConstantForceField.inl

```diff
@@ -92,30 +91,36 @@ template<class DataTypes>
 void ConstantForceField<DataTypes>::init()
 {
     this->m_componentstate = core::objectmodel::ComponentState::Invalid;
-
-    if (l_topology.empty())
+       
+    std::string templateName = this->templateName(this);
+    if (templateName.find("Rigid") == std::string::npos)
     {
-        msg_warning() << "link to Topology container should be set to ensure right behavior. First Topology found in current context will be used.";
-        l_topology.set(this->getContext()->getMeshTopology());
-    }
+        if (l_topology.empty())
+        {
+            msg_warning() << "link to Topology container should be set to ensure right behavior. First Topology found in current context will be used.";
```

Comment:
Hey @epernod 

I think this should be a msg_info(). For me, there's nothing wrong in using the scene graph to automatically create links between components. The important thing was to let the user the ability to explicitly see the links and change them if he wants to.

If we put msg_warning, that will mean we have to specify the paths everywhere, which kind of defeat the purpose of the scene graph. If we don't do this, than we will be flooded with warning message.

What do you think?

Reply:
Small addition: We could send the msg_info() writing the path of the `this->getContext()->getMeshTopology()` if there is one. This way the user will be able to look at the log if something goes wrong, and than see that the path isn't the good one. If there isn't any topology found, than a warning_msg or error_msg could be sent.

Reply:
I fully agree with @jnbrunet comment.

Reply:
ok. it's fine for me. Thanks for this convergence. I'll do the change.

Reply:
But I will open an issue to keep the list of component that shows an error on the CI for this PR. They reveal all the components that have a pointer to the topology and not really using it or in a tricky way.